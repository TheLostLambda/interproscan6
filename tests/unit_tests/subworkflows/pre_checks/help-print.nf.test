
nextflow_function {

    name "Test Functions"
    script "interproscan/subworkflows/pre_checks/main.nf"

    stage {
        symlink "interproscan/scripts/pre_checks/check_nucleic_seq.py"
        symlink "interproscan/modules/pre_checks/main.nf"
    }

    test("Test Function printHelp") {

        function "printHelp"

        when {
            function {
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }
}