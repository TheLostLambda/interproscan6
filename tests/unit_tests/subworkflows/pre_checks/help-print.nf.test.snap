{
    "Test Function printHelp": {
        "content": [
            "\n    Usage example:\n        nextflow run interproscan.nf --input <path to fasta file>\n\n    Params options:\n        --applications <ANALYSES>          Optional, comma separated - without spaces - list of analysis methods (i.e. member databases/applications).\n                                            If this option is not set, ALL analyses will be run.\n        --disable_precalc                  Optional. Disables use of the precalculated match lookup service.\n                                            All match calculations will be run locally.\n        --formats <FORMATS>                Optional, comma separated - without spaces - list of output formats.\n        --goterms                          Optional. Include GO terms in the output.\n        --help                             Optional, display help information\n        --input <INPUT-FILE-PATH>          [REQUIRED] Path to fasta file that should be loaded on Master startup.\n        --nucleic                          Optional. Input comprises nucleic acid sequences.\n        --outdir <OUTPUT-DIR-PATH>         Optional. Path to the output dir.\n                                            Output files are automatically named after the input file, with the suffix '.ips6.*'. Default: present working dir.\n        --pathways                         Optional. Include pathway information in the output.\n        --signalp_mode                     Optional. Set which SignalP/SignalP_EUK prediction models are used. Models may have to be installed. Accepted: 'fast', 'slow', 'slow-sequential'. Default: 'fast'.\n        --version                          Print the version of InterProScan.\n"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-10T17:47:22.355126"
    }
}
